lib/src/iterable_extensions.dart

```diff
@@ -601,11 +638,7 @@ extension IterableNullableExtension<T extends Object> on Iterable<T?> {
   /// of this iterable, in their original iteration order.
   ///
   /// For an `Iterable<X?>`, this method is equivalent to `.whereType<X>()`.
-  Iterable<T> whereNotNull() sync* {
-    for (var element in this) {
-      if (element != null) yield element;
-    }
-  }
+  Iterable<T> whereNotNull() => _NotNullIterable(this);
```

Comment:
This is a very important check, since it transform a new `Iterable` instance (_NotNullIterable) to just a casting.

Since the extension is resolved in the variable level, not in the instance:

```dart
void main() {
  
  List<int?> l1 = <int>[1,2];
  List<int> l2 = <int>[3,4];
  List<int?> l3 = <int?>[1,2];
  
  print(l1.isNullable);
  print(l2.isNullable);
  print(l3.isNullable);
  
}

extension _ItrExt<T extends Object> on Iterable<T> {
  String get isNullable => "NO<T>";
}

extension _ItrNullableExt<T extends Object> on Iterable<T?> {
  String get isNullable => this is Iterable<T> ? "NO<T?>" : "YES<T?>";
}

/////////////////
// OUTPUT:
/////////////////
// NO<T?>
// NO<T>
// YES<T?>
```